src/components/NewDatePicker/index.js

```diff
@@ -40,66 +44,58 @@ const datePickerDefaultProps = {
     value: undefined,
 };
 
-class NewDatePicker extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            selectedDate: props.value || props.defaultValue || undefined,
-        };
+function NewDatePicker(props) {
+    const [date, setDate] = useState(props.value || props.defaultValue || undefined);
```

Comment:
I think `selectedDate` was more specific and descriptive here
```suggestion
    const [selectedDate, setSelectedDate] = useState(props.value || props.defaultValue || undefined);
```